### DISCUSSION TITLE
How exit lauched clangd program from lsp request?

### DISCUSSION THREAD

**Discussion by Yurii-huang** [clangd/clangd#1153] [Q&A] [answered]
clangd version
```
clangd version 13.0.0 (https://github.com/llvm/llvm-project d7b669b3a30345cfcdb2fde2af6f48aa4b94845d)
Features: linux+grpc
Platform: x86_64-unknown-linux-gnu

```

for protocol log:
``` bash
Request ->  {
    "id": 2,
    "jsonrpc": "2.0",
    "method": "shutdown"
}

Response <-  {"id":2,"jsonrpc":"2.0","result":null}

Request ->  {
    "id": 3,
    "jsonrpc": "2.0",
    "method": "exit",
    "params": null
}

Response <-  {"error":{"code":-32601,"message":"method not found"},"id":3,"jsonrpc":"2.0"}
```

It doesn't seem to work，please help me，thank anyone.

**Comment by HighCommander4** [comment, marked as answer]
I think the issue is that ["exit"](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#exit) is a ["notification message"](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#notificationMessage).

Unlike a ["request message"](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#requestMessage), a "notification message" should not contain an "id" field.

**Reply by Yurii-huang** [reply]
Sorry, this question seems a little retarded.


In fact, I want to know how to exit the started Clangd, because I tried all kinds of methods, but they didn't seem to work. My Clangd is started through stdio. I want to know how I can exit normally in the standard protocol, rather than end by killing the process.


I want to confirm that exit is supported by Clangd

**Reply by HighCommander4** [reply]
I understand the question. I think your issue is the one I pointed out. Your "exit" request:

```
Request ->  {
    "id": 3,
    "jsonrpc": "2.0",
    "method": "exit",
    "params": null
}
```

contains an "id" field when it's not supposed to. Try removing the "id" field from that message and it should work.

**Reply by Yurii-huang** [reply]
Thank you.
I'm sure you're Superman. 
It really works.
